### ISSUE TITLE
SIGFPE on startup

### ISSUE THREAD

**Issue by SirCmpwn** [Alexays/Waybar#9]
Just built from source using the default config and gave it a run:

```
(gdb) bt full
#0  0x00005566cf88b4ff in waybar::modules::Battery::update() (this=0x5566cfeba460) at ../src/modules/battery.cpp:46
        total = 0
        charging = false
        format = "{}% "
#1  0x00005566cf88aa91 in waybar::modules::Battery::<lambda()>::operator()(void) const (__closure=0x5566cfedf7a0) at ../src/modules/battery.cpp:20
        this = 0x5566cfeba460
#2  0x00005566cf88b974 in std::_Function_handler<void(), waybar::modules::Battery::Battery(Json::Value)::<lambda()> >::_M_invoke(const std::_Any_data &) (__functor=...)
    at /usr/include/c++/8.2.0/bits/std_function.h:297
#3  0x00005566cf88d88e in std::function<void ()>::operator()() const (this=0x5566cfedf7a0) at /usr/include/c++/8.2.0/bits/std_function.h:687
#4  0x00005566cf88c7eb in waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}::operator()() const (__closure=0x5566cfedf798) at ../include/util/chrono.hpp:48
        func = 
          {<std::_Maybe_unary_or_binary_function<void>> = {<No data fields>}, <std::_Function_base> = {static _M_max_size = 16, static _M_max_align = 8, _M_functor = {_M_unused = {_M_object = 0x5566cfeba460, _M_const_object = 0x5566cfeba460, _M_function_pointer = 0x5566cfeba460, _M_member_pointer = (void (std::_Undefined_class::*)(std::_Undefined_class * const)) 0x5566cfeba460, this adjustment 140723131598032}, _M_pod_data = "`\244\353\317fU\000\000\320DE\250\374\177\000"}, _M_manager = 0x5566cf88b977 <std::_Function_base::_Base_manager<waybar::modules::Battery::Battery(Json::Value)::<lambda()> >::_M_manager(std::_Any_data &, const std::_Any_data &, std::_Manager_operation)>}, _M_invoker = 0x5566cf88b954 <std::_Function_handler<void(), waybar::modules::Battery::Battery(Json::Value)::<lambda()> >::_M_invoke(const std::_Any_data &)>}
        this = 0x5566cfeba480
#5  0x00005566cf88eb01 in std::__invoke_impl<void, waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}>(std::__invoke_other, waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}&&) (__f=...) at /usr/include/c++/8.2.0/bits/invoke.h:60
#6  0x00005566cf88d94b in std::__invoke<waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}>(std::__invoke_result&&, (waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}&&)...) (__fn=...) at /usr/include/c++/8.2.0/bits/invoke.h:95
#7  0x00005566cf8911b2 in std::thread::_Invoker<std::tuple<waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) (this=0x5566cfedf798) at /usr/include/c++/8.2.0/thread:234
#8  0x00005566cf891188 in std::thread::_Invoker<std::tuple<waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}> >::operator()() (this=0x5566cfedf798)
    at /usr/include/c++/8.2.0/thread:243
#9  0x00005566cf89116c in std::thread::_State_impl<std::thread::_Invoker<std::tuple<waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}> > >::_M_run() (this=0x5566cfedf790) at /usr/include/c++/8.2.0/thread:186
#10 0x00007f08aa739043 in execute_native_thread_routine (__p=0x5566cfedf790) at /build/gcc/src/gcc/libstdc++-v3/src/c++11/thread.cc:80
        __t = std::unique_ptr<std::thread::_State> = {get() = 0x5566cfedf790}
#11 0x00007f08aa4c4a8d in start_thread () at /usr/lib/libpthread.so.0
#12 0x00007f08aa3f4823 in clone () at /usr/lib/libc.so.6
```

Note, I don't have a battery.

**Comment by Alexays** [comment]
Hmm thats why :smile:  
There is no check when no batteries is available
Fixed in d728de2dd7534dc22a545da57445d86d9abaddbf

**Comment by SirCmpwn** [comment]
Thanks!
